deepchem/models/tensorgraph/layers.py

```diff
@@ -86,12 +98,35 @@ def shared(self, in_layers):
     -------
     Layer
     """
+    if tfe.in_eager_mode():
+      raise ValueError('shared() is not supported in eager mode')
     if self.variable_scope == '':
       return self.clone(in_layers)
     raise ValueError('%s does not implement shared()' % self.__class__.__name__)
 
-  def __call__(self, *in_layers, **kwargs):
-    return self.create_tensor(in_layers=in_layers, set_tensors=False, **kwargs)
+  def __call__(self, *inputs, **kwargs):
+    """Execute the layer in eager mode to compute its output as a function of inputs.
```

Comment:
Users might want to use the `__call__` syntax in graph mode as well, especially as they become more familiar with using it.